tee-worker/litentry/core/vc-task/receiver/src/mock.rs

```diff
@@ -0,0 +1,229 @@
+use itp_stf_executor::mocks::StfEnclaveSignerMock;
+use itp_stf_primitives::{traits::TrustedCallSigning, types::KeyPair};
+use itp_test::mock::{
+	handle_state_mock::HandleStateMock, onchain_mock::OnchainMock,
+	shielding_crypto_mock::ShieldingCryptoMock,
+};
+use itp_top_pool_author::{
+	error::Result,
+	mocks::{AuthorApiMock, GLOBAL_MOCK_AUTHOR_API, GLOBAL_MOCK_RPC_API},
+};
+use itp_types::MrEnclave;
+use lazy_static::lazy_static;
+use lc_data_providers::DataProviderConfig;
+use lc_stf_task_sender::{AssertionBuildRequest, RequestType};
+use litentry_primitives::{aes_decrypt, AesRequest, Assertion, RequestAesKey};
+use sp_core::{blake2_256, sr25519, Pair};
+use std::{sync::mpsc::Receiver, vec::Vec};
+
+pub const COMMON_SEED: &[u8] =
+	b"crouch whisper apple ladder skull blouse ridge oven despair cloth pony";
+lazy_static! {
+	pub static ref SHIELDING_KEY: ShieldingCryptoMock = ShieldingCryptoMock::default();
+}
+
+pub fn init_global_mock_author_api() -> Result<Receiver<Vec<u8>>> {
+	let (sender, receiver) = std::sync::mpsc::channel();
+	let mut stf_task_storage = GLOBAL_MOCK_AUTHOR_API.lock().unwrap();
+	*stf_task_storage = Some(sender);
+	Ok(receiver)
+}
+
+pub fn init_global_rpc_api() -> Result<Receiver<Vec<u8>>> {
+	let (sender, receiver) = std::sync::mpsc::channel();
+	let mut stf_task_storage = GLOBAL_MOCK_RPC_API.lock().unwrap();
+	*stf_task_storage = Some(sender);
+	Ok(receiver)
+}
+
+pub fn init_stf_context() -> StfTaskContext<
+	KeyRepositoryMock<ShieldingCryptoMock>,
+	AuthorApiMock<H256, H256, TrustedCallSigned, Getter>,
+	StfEnclaveSignerMock,
+	HandleStateMock,
+	OnchainMock,
+> {
+	let shielding_key = SHIELDING_KEY.clone();
+	let shielding_key_repository_mock = KeyRepositoryMock::new(shielding_key.clone());
+	let author_mock = AuthorApiMock::default();
+	let stf_enclave_signer_mock = StfEnclaveSignerMock::default();
+	let handle_state_mock = HandleStateMock::from_shard(H256::zero()).unwrap();
+	let onchain_mock = OnchainMock::default();
+	let data_provider_conifg = DataProviderConfig::new().unwrap();
+
+	let context = StfTaskContext::new(
+		Arc::new(shielding_key_repository_mock),
+		author_mock.into(),
+		stf_enclave_signer_mock.into(),
+		handle_state_mock.into(),
+		onchain_mock.into(),
+		data_provider_conifg.into(),
+	);
+	context
+}
+
+pub fn init_vc_task() {
+	let stf_context = init_stf_context();
+	let node_metadata = NodeMetadataRepository::<NodeMetadataMock>::new(NodeMetadataMock::new());
+	let extrinsic_factory = ExtrinsicsFactoryMock::default();
+	std::thread::spawn(move || {
+		run_vc_handler_runner(
+			Arc::new(stf_context),
+			Arc::new(extrinsic_factory),
+			Arc::new(node_metadata),
+		);
+	});
+}
+
+pub fn execute_with_vc_task<F>(process_fn: F)
+where
+	F: FnOnce(&Receiver<Vec<u8>>, &Receiver<Vec<u8>>),
+{
+	init_vc_task();
+	let top_calls_receiver = init_global_mock_author_api().unwrap();
+	let rpc_calls_receiver = init_global_rpc_api().unwrap();
+
+	thread::sleep(Duration::from_secs(2));
```

Comment:
What's the purpose of this `sleep` ?

Reply:
This is a sanity check to make sure things are initialised.